src/test/java/jp/co/yahoo/yosegi/message/formatter/json/TestJacksonArrayFormatter.java

```diff
@@ -0,0 +1,358 @@
+
+    @Override
+    public boolean containsKey( final String key ) throws IOException {
+      if ( "key1".equals( key ) ) {
+        return true;
+      } else if ( "key2".equals( key ) ) {
+        return true;
+      } else if ( "key3".equals( key ) ) {
+        return true;
+      }
+      return false;
+    }
+
+    @Override
+    public int size() throws IOException {
+      return 3;
+    }
+
+    @Override
+    public boolean isArray() throws IOException {
+      return isArray;
+    }
+
+    @Override
+    public boolean isMap() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean isStruct() throws IOException {
+      return ! isArray;
+    }
+
+    @Override
+    public boolean hasParser( final int index ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public boolean hasParser( final String key ) throws IOException {
+      return false;
+    }
+
+    @Override
+    public Object toJavaObject() throws IOException {
+      return null;
+    }
+
+  }
+
+  @Test
+  public void T_write_1() throws IOException {
+    List<Object> list = Arrays.asList( "a" , "b" , "c" );
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( list );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode arrayNode = (ArrayNode)node;
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get(0).isTextual() );
+    assertTrue( arrayNode.get(1).isTextual() );
+    assertTrue( arrayNode.get(2).isTextual() );
+    assertEquals( arrayNode.get(0).asText() , "a" );
+    assertEquals( arrayNode.get(1).asText() , "b" );
+    assertEquals( arrayNode.get(2).asText() , "c" );
+  }
+
+  @Test
+  public void T_write_3() throws IOException {
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( null );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode arrayNode = (ArrayNode)node;
+    assertEquals( arrayNode.size() , 0 );
+  }
+
+  @Test
+  public void T_write_4() throws IOException {
+    List<Object> list = Arrays.asList( Arrays.asList( "a" , "b" , "c" ) );
+    JacksonArrayFormatter formatter = new JacksonArrayFormatter();
+    JsonNode node = formatter.write( list );
+    assertTrue( ( node instanceof ArrayNode ) );
+
+    ArrayNode parentNode = (ArrayNode)node;
+    assertEquals( parentNode.size() , 1 );
+
+    JsonNode childNode = parentNode.get(0);
+    assertTrue( ( childNode instanceof ArrayNode ) );
+    ArrayNode arrayNode = (ArrayNode)childNode;
+
+    assertEquals( arrayNode.size() , 3 );
+    assertTrue( arrayNode.get(0).isTextual() );
+    assertTrue( arrayNode.get(1).isTextual() );
+    assertTrue( arrayNode.get(2).isTextual() );
+    assertEquals( arrayNode.get(0).asText() , "a" );
+    assertEquals( arrayNode.get(1).asText() , "b" );
+    assertEquals( arrayNode.get(2).asText() , "c" );
+  }
+
+  @Test
+  public void T_write_5() throws IOException {
```

Comment:
Please write the test method name is to describe what is wanted to in the test instead of using number (it makes vague one).

Reply:
I changed the method name.